scripts/benchmark_multithreaded.py

```diff
@@ -0,0 +1,205 @@
+
+def get_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-v', '--verbose', action='store_true', help='Print log messages')
+    parser.add_argument('-q', '--queries', action='store', type=int, metavar='Q', nargs='+', help='Specify the TPC-H queries that will be benchmarked')
+    parser.add_argument('--result-dir', action='store', type=str, metavar='DIR', default='results', help='Directory where the result folder will be stored (default: \'results/\')')
+    parser.add_argument('--result-name', action='store', type=str, metavar='NAME', help='Directory where the actual results will be stored (default: current datetime)')
+    parser.add_argument('executable', action='store', type=str, metavar='EXECUTABLE', help='hyriseBenchmarkTPCH executable')
+    parser.add_argument('--clients', action='store', type=int, metavar='C', default=10, help='The number of clients that schedule queries in parallel')
+
+    core_group = parser.add_mutually_exclusive_group()
+    core_group.add_argument('-c', '--cores', action='store', type=int, metavar='C', nargs='+', help='List of cores to be used for the benchmarks')
+    core_group.add_argument('--max-cores', action='store', type=int, metavar='C', help='Number of cores up to which the benchmarks will be executed')
+
+    run_group = parser.add_mutually_exclusive_group(required=True)
+    run_group.add_argument('-r', '--runs', action='store', type=int, metavar='N', help='Fixed number of (max) runs each query is executed')
+    run_group.add_argument('--runs-per-core', action='store', type=int, metavar='N', help='Number of (max) runs per core each query is executed')
+    
+    return parser
+
+def verbose_print(verbose, message):
+    if verbose: print('python> ' + message)
+
+def get_core_counts(args):
+    if args.cores:
+        return args.cores
+    max_core_count = args.max_cores if args.max_cores else MAX_CORE_COUNT
+    if max_core_count < 8:
+        step = -1
+    elif max_core_count < 16:
+        step = -int(max_core_count / 4)
+    else:
+        step = -int(max_core_count / 8)
+    core_counts = list(range(max_core_count, 0, step))
+    if core_counts[-1] != 1:
+        core_counts.append(1)
+    core_counts.append(0)
+    return core_counts
+
+def get_formatted_queries(args):
+    queries = args.queries if args.queries else DEFAULT_TPCH_QUERIES
+    query_ids = ','.join([str(query) for query in queries])
+    return ['--queries', query_ids]
+
+def is_square(n):
+    return n ** 0.5 == int(n ** 0.5)
+
+def get_subplot_row_and_column_count(num_plots):
+    while not is_square(num_plots):
+        num_plots += 1
+    return num_plots ** 0.5
+
+def run_benchmarks(args, hyrise_args, core_counts, result_dir):
+    benchmark_run = 0
+    for core_count in core_counts:
+        benchmark_run += 1
+        verbose_print(args.verbose, 'Starting benchmark run ' + str(benchmark_run) + ' of ' + str(len(core_counts)) + ' ...')
```

Comment:
I would leave out "run" here. In the context of benchmarks, I understand it as if one benchmark was executed multiple times.